Add support for GPTQ using native transformers/peft

winglian opened the pull request.

winglian committed:
auto gptq support

winglian committed:
more tweaks and add yml

winglian committed:
remove old gptq docker

winglian committed:
don't need explicit peft install for tests

NanoCode012 commented on examples/gptq/lora.yml:
```diff
@@ -0,0 +1,76 @@
+base_model: TheBloke/Llama-2-7B-GPTQ
```
this file should maybe be in `examples/llama2/gptq-lora.yml`

NanoCode012 commented on requirements.txt:
```diff
@@ -1,3 +1,5 @@
+--extra-index-url https://huggingface.github.io/autogptq-index/whl/cu117/
```
is this cuda version ok?

NanoCode012 commented on src/axolotl/utils/models.py:
```diff
@@ -147,32 +148,15 @@ def load_model(
     if cfg.model_revision:
         model_kwargs["revision"] = cfg.model_revision
+    if cfg.gptq:
+        # TODO we should figure out how read the models config.json first
```
We can load from config with AutoConfig.from_pretrained?

winglian replied:
yeah, I might tackle this in a refactor. not sure how easily this is handled for cases where it points to a local model path.

winglian committed:
fix setup.py to use extra index url

install torch for tests
fix cuda version for autogptq index
set torch in requirements so that it installs properly
move gptq install around to work with github cicd

winglian committed:
gptq doesn't play well with sample packing

winglian committed:
address pr feedback

winglian committed:
remove torch install for now

winglian committed:
set quantization_config from model config

winglian committed:
Fix the implementation for getting quant config from model config
